src/app/action-items/action-items.component.spec.ts

```diff
@@ -35,42 +27,38 @@ describe('ActionItemsComponent', () => {
     expect(component).toBeTruthy();
   });
 
-  it('should stop loading and render the view after 1s', done => {
-    component.loading = true;
+  it('should return an array with 1 object', async(() => {
+    const response: ActionTasksElement[] = [];
+    spyOn(tasksService, 'getAllTasks').and.returnValue(of(response));
+    component.getTasks();
     component.ngOnInit();
-    setTimeout(() => {
-      expect(component.loading).toBe(false);
-      done();
-    }, 1000);
-  });
-
-  it('should return an array with 1 object', () => {
-    const tasksService = new TasksService();
-    spyOn(tasksService, 'getAllTasks').and.returnValue(of(tableData));
-    tasksService.getAllTasks();
-    expect(tableData.length).toBe(1);
-  });
+    expect(typeof component.dataSource).toBe(typeof response);
```

Comment:
As dataSource is typed now, then what's the need of such test?
Btw. if all tests have the same setup then it's worth to move the common logic into before each or just merge it into the single test (if there's the only difference with expects).